### PR TITLE
Add test cases for getMatchingWord()

### PR DESCRIPTION
**Pull request by nigma456**
##### Objective
Added test cases for getMatchingWord() in the autocompletion.test.ts
##### Abstractions
This function is used by the autocompletion feature and it is not covered before since we wrongly used the readOnlyDocument class which is defined only for project feature use. 
There are some limitation for this readOnlyDocument class, like it does not implement the getWordRangeAtPosition() method which prevent writing the getMatchingWord test cases before.

In this submission I simply implement the function readFile2TextDocument() which will read a file to a vscode.TextDocument object so that we can use it later.
In most of our unit test cases we should  use this readFile2TextDocument() methd instead of readOnlyDocument.

2. Delete used folder /test_cases

### CODE REVIEW

File: extension/src/test/suite/helper.ts [hunk 1]
```diff
@@ -10,3 +11,12 @@ export function princ(str : any){
     var value = str[name];
 	console.log(`-----the ${name} is ${value.toString()}-----`);
 }
+//This method is to read a file and return the vscode.TextDocument object 
+export async function readFile2TextDocument(filepath: string) : Promise<vscode.TextDocument | undefined> {
+	let doc : vscode.TextDocument;
```
Comment (Sen-real) [inline extension/src/test/suite/helper.ts:16]: this variable should be removed as the "Code scanning" suggests

File: extension/src/test/suite/helper.ts [hunk 2]
```diff
@@ -10,3 +11,12 @@ export function princ(str : any){
...
+	try {
+        return doc = await vscode.workspace.openTextDocument(filepath);
+      } catch (error) {
+        return undefined;
```
Comment (Sen-real) [inline extension/src/test/suite/helper.ts:20]: how about console.log(error) before return?

**Review by Sen-real — APPROVED** [review]

File: extension/src/test/suite/helper.ts (hunk 2 above)
Comment (nigma456) [reply extension/src/test/suite/helper.ts:20]: Yes, I will add this.

File: extension/src/test/suite/helper.ts (hunk 1 above)
Comment (nigma456) [reply extension/src/test/suite/helper.ts:16]: Will do a minor change to keep this variable.

**Review by Sen-real — APPROVED** [review]

**Pull request merged**